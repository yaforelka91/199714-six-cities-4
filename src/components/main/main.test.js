import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import Main from './main.jsx';

const mockStore = configureStore([]);

const offersList = [
  {
    id: 0,
    coords: [52.3909553943508, 4.85309666406198],
    title: `Beautiful & luxurious apartment at great location`,
    description: [
      `A quiet cozy and picturesque that hides behind a 
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
      `An independent House, strategically located 
  between Rembrand Square and National Opera, 
  but where the bustle of the city comes to rest 
  in this alley flowery and colorful.`
    ],
    picture: `http://placeimg.com/260/200/arch`,
    pictures: [
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`
    ],
    price: 120,
    type: `Apartment`,
    isPremium: true,
    rating: 4.1,
    bedrooms: `3 Bedrooms`,
    guests: `Max 4 adults`,
    services: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      picture: `http://placekitten.com/74/74`,
      isSuper: true,
    },
    reviews: [`0`, `1`],
  },
];

const cities = [
  {
    id: 1,
    name: `city 1`,
    coords: [0, 0],
  },
];

describe(`MainSnapshot`, () => {
  it(`should render correctly`, () => {
    const store = mockStore({
      activeSorting: `popular`,
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Main
            activeSorting='popular'
            offersList={offersList}
            citiesList={cities}
            city={cities[0]}
            onOfferTitleClick={() => {}}
            onCityNameClick={() => {}}
          />
        </Provider>,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render placeholder if offers array is empty`, () => {
    const store = mockStore({
      activeSorting: `popular`,
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Main
            activeSorting='popular'
            offersList={[]}
            citiesList={cities}
            city={cities[0]}
            onOfferTitleClick={() => {}}
            onCityNameClick={() => {}}
          />
        </Provider>,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
