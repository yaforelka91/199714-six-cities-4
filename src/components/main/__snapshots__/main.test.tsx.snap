// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSnapshot should render correctly 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="/"
            onClick={[Function]}
          >
            <span>
              city 1
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          1
           places to stay in 
          city 1
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
           
          <span
            className="places__sorting-type"
            onClick={[Function]}
            tabIndex={0}
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom"
          >
            <li
              className="places__option places__option--active"
              onClick={[Function]}
              tabIndex={0}
            >
              Popular
            </li>
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex={0}
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex={0}
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex={0}
            >
              Top rated first
            </li>
          </ul>
          <select
            className="places__sorting-type"
            id="places-sorting"
            onChange={[Function]}
            style={
              Object {
                "border": 0,
                "clip": "rect(1px, 1px, 1px, 1px)",
                "height": "1px",
                "overflow": "hidden",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            value="popular"
          >
            <option
              className="places__option"
              value="popular"
            >
              Popular
            </option>
            <option
              className="places__option"
              value="to-high"
            >
              Price: low to high
            </option>
            <option
              className="places__option"
              value="to-low"
            >
              Price: high to low
            </option>
            <option
              className="places__option"
              value="top-rated"
            >
              Top rated first
            </option>
          </select>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        >
          <article
            className="cities__place-card place-card"
            onMouseEnter={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/0"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="http://placeimg.com/260/200/arch"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    120
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/0"
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment at great location
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;

exports[`MainSnapshot should render placeholder if offers array is empty 1`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="/"
            onClick={[Function]}
          >
            <span>
              city 1
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container cities__places-container--empty"
    >
      <section
        className="cities__no-places"
      >
        <div
          className="cities__status-wrapper tabs__content"
        >
          <b
            className="cities__status"
          >
            No places to stay available
          </b>
          <p
            className="cities__status-description"
          >
            We could not find any property availbale at the moment in city 1
          </p>
        </div>
      </section>
      <div
        className="cities__right-section"
      />
    </div>
  </div>
</main>
`;
